### PR TITLE
 Update operation for AzDO-based Git Repository resource

### PR DESCRIPTION
**Pull request by TsuyoshiUshio**
## All Submissions:
-------------------------------------
* [YES] Have you added an explanation of what your changes do and why you'd like us to include them?
* [NO] I have updated the documentation accordingly.
* [YES] I have added tests to cover my changes.
* [YES] All new and existing tests passed.
* [YES] My code follows the code style of this project.
* [YES] I ran lint checks locally prior to submission.
* [YES] Have you checked to ensure there aren't other open PRs for the same update/change?

## What is the current behavior?
-------------------------------------

https://github.com/microsoft/terraform-provider-azuredevops/issues/97

## What is the new behavior?
-------------------------------------

Update the Repository

## Does this introduce a breaking change?
-------------------------------------
- [NO]

Just add a new feature.

## Other information

I'm wondering if this is correct way to write unit testing, could you review it? 

The challenge is, in case of update, we need to write the condition if the mock override the return. In case of update, it will be object.  the object includes several parameters, it includes pointers. In this case, even if the value is the same, it is assumed that it is different one. 

Since the test is a test for REST API call fail case, so I just use gomock.Any() as the expected parameter.

### CODE REVIEW

**Comment by TsuyoshiUshio** [comment]
BTW, I found one weird behavior about linting. When I run 'scirpt/lint-check-go.sh` it report 0, however, Ci's lint report 1 issue. I switch back this line to original (it was modified by VSCode) then locally, lint report 1, CI success.  The point is here. 

```
	aggregatedClient := &aggregatedClient{
		CoreClient:            coreClient,
		BuildClient:           buildClient,
		GitReposClient:        gitReposClient,
		OperationsClient:      operationsClient,
		ServiceEndpointClient: serviceEndpointClient,
		ctx:                   ctx,     <--------- this 
	}
```

File: azuredevops/resource_azure_git_repository_test.go [hunk 1]
```diff
@@ -213,7 +235,8 @@ func TestAzureGitRepo_Read_UsesNameIfIdNotSet(t *testing.T) {
 //	(5) resource can no longer be queried by ID
 func TestAccAzureGitRepo_Create(t *testing.T) {
```
Comment (nmiodice) [inline azuredevops/resource_azure_git_repository_test.go]:
```suggestion
func TestAccAzureGitRepo_CreateAndUpdate(t *testing.T) {
```

**Comment by nmiodice** [comment]
@TsuyoshiUshio regarding the linting issue, I am seeing the same behavior on my end. Here is my re-creation steps:

```bash
$ git checkout -b TsuyoshiUshio-feature/updategitrepo master
$ git pull https://github.com/TsuyoshiUshio/terraform-provider-azuredevops.git feature/updategitrepo
$ ./scripts/lint-check-go.sh 
[INFO] Linting Go Files... If this fails, run 'find . -name "*.go" | xargs gofmt -s -w' to fix
[INFO] Found 0 with linting issues...
$ go version
go version go1.12.10 linux/amd64
```

Initially I thought it was related to the version of Go. but this matches our CI/CD environment. Did you find any clues as to what the root cause is?

**Comment by nmiodice** [comment]
Actually, on second thought, I was on the wrong commit! I am able to get a linting error from the relevant commit... Perhaps it is your version of Go or possibly something related to VS code?

```bash
$ git checkout 6f31b5f43b38557c5091b46461ffaf8818d200f7
$ ./../scripts/lint-check-go.sh 
[INFO] Linting Go Files... If this fails, run 'find . -name "*.go" | xargs gofmt -s -w' to fix
[INFO] Found 1 with linting issues...
```

File: azuredevops/resource_azure_git_repository_test.go (hunk 1 above)
Comment (TsuyoshiUshio) [reply azuredevops/resource_azure_git_repository_test.go]: Oh! silly mistake lol. I fixed that. :)

**Comment by TsuyoshiUshio** [comment]
@nmiodice I merged. Thanks. for the linitng one, I misunderstand the shell responsibility. The lint shell script is not just lint chiking, but also format the code. :) That is why If I execute the linitng, it will fix. I see. :) I should have read code first. lol

**Pull request merged**